src/calendar.cpp

```diff
@@ -499,36 +499,19 @@ std::string calendar::day_of_week() const
         WEDNESDAY = 6
     };
 
-    // calendar::day gets mangled by season transitions, so recalculate days since start.
-    int current_day = turn_number / DAYS(1) % 7;
-
-    std::string day_string;
-
-    switch (current_day) {
-    case SUNDAY:
-        day_string = _("Sunday");
-        break;
-    case MONDAY:
-        day_string = _("Monday");
-        break;
-    case TUESDAY:
-        day_string = _("Tuesday");
-        break;
-    case WEDNESDAY:
-        day_string = _("Wednesday");
-        break;
-    case THURSDAY:
-        day_string = _("Thursday");
-        break;
-    case FRIDAY:
-        day_string = _("Friday");
-        break;
-    case SATURDAY:
-        day_string = _("Saturday");
-        break;
-    }
+    static const std::map<weekday, std::string> weekday_names = {
+        std::pair<weekday, std::string>{THURSDAY, translate_marker( "Thursday" )},
+        std::pair<weekday, std::string>{FRIDAY, translate_marker( "Friday" )},
+        std::pair<weekday, std::string>{SATURDAY, translate_marker( "Saturday" )},
+        std::pair<weekday, std::string>{SUNDAY, translate_marker( "Sunday" )},
+        std::pair<weekday, std::string>{MONDAY, translate_marker( "Monday" )},
+        std::pair<weekday, std::string>{TUESDAY, translate_marker( "Tuesday" )},
+        std::pair<weekday, std::string>{WEDNESDAY, translate_marker( "Wednesday" )}
+    };
 
-    return day_string;
+    // calendar::day gets mangled by season transitions, so recalculate days since start.
+    const weekday current_day = static_cast< weekday >( turn_number / DAYS(1) % 7 );
```

Comment:
The line makes at least two assumptions (both are pretty solid):
- There are exactly seven days of the week;
- Their indexes are continuous.

These are perfect match for a simple `array`. Also, a side note: the rationale (up there), explaining why Thursday is the day zero, is nice to know, but IMO it shouldn't affect the `enum`. I'd change the order to be more old-fashioned:

```c++
enum weekday {
    SUNDAY = 0,
    MONDAY,
    TUESDAY,
    WEDNESDAY,
    THURSDAY,
    FRIDAY,
    SATURDAY
};

static const std::array<std::string, 7> weekday_names = {
	translate_marker( "Sunday" ),
	translate_marker( "Monday" ),
	translate_marker( "Tuesday" ),
	translate_marker( "Wednesday" ),
	translate_marker( "Thursday" ),
	translate_marker( "Friday" ),
	translate_marker( "Saturday" ),	
};

const weekday day_zero = THURSDAY;
const size_t current_day = ( turn_number / DAYS( 1 ) + day_zero ) % weekday_names::size();

```